src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs

```diff
@@ -18,18 +33,92 @@ namespace Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
+        private readonly ITextView _textView;
+        private readonly SnapshotSpan _span;
+        private readonly SymbolKey _key;
+        private readonly IThreadingContext _threadingContext;
+        public readonly Lazy<IStreamingFindUsagesPresenter> StreamingPresenter;
+
+        private InlineParameterNameHintsTag(FrameworkElement adornment, ITextView textView, SnapshotSpan span,
+                                            SymbolKey key, InlineParameterNameHintsTaggerProvider taggerProvider)
+            : base(adornment, removalCallback: null, PositionAffinity.Predecessor)
+        {
+            _textView = textView;
+            _span = span;
+            _key = key;
+            StreamingPresenter = taggerProvider.StreamingFindUsagesPresenter;
+            _threadingContext = taggerProvider.ThreadingContext;
+            _toolTipService = taggerProvider.ToolTipService;
+
+            // Sets the tooltip to a string so that the tool tip opening event can be triggered
+            // Tooltip value does not matter at this point
+            adornment.ToolTip = "Quick info";
+            adornment.ToolTipOpening += Border_ToolTipOpening;
+        }
 
         /// <summary>
         /// Creates the UIElement on call
         /// Uses PositionAffinity.Predecessor because we want the tag to be associated with the preceding character
         /// </summary>
-        /// <param name="text">The name of the parameter associated with the argument</param>
-        public InlineParameterNameHintsTag(string text, double lineHeight, TextFormattingRunProperties format)
-            : base(CreateElement(text, lineHeight, format), removalCallback: null, PositionAffinity.Predecessor)
+        /// <param name="textView">The view of the editor</param>
+        /// <param name="span">The span that has the location of the hint</param>
+        /// <param name="key">The symbolkey associated with each parameter</param>
+        public static InlineParameterNameHintsTag Create(string text, double lineHeight, TextFormattingRunProperties format,
+                                                         ITextView textView, SnapshotSpan span, SymbolKey key,
+                                                         InlineParameterNameHintsTaggerProvider taggerProvider)
+        {
+            return new InlineParameterNameHintsTag(CreateElement(text, lineHeight, format), textView,
+                                                   span, key, taggerProvider);
+        }
+
+        public async Task<IReadOnlyCollection<object>> CreateDescriptionAsync(CancellationToken cancellationToken)
         {
+            var document = _textView.TextBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            var textContentBuilder = new List<TaggedText>();
+
+            if (document != null)
+            {
+                var compilation = await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Small nit: call GetRequiredCompilationAsync, as this will flag when my PR goes in. Your code is fine (we're not going to get a null compilation here) but this will prevent a potential broken build depending on which order we merge in.

Reply:
Changed